microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,495 @@
+
+Widgets are also available as an origin trial in Microsoft Edge 108. Learn [how to enroll your site in an origin trial](origin-trials.md#enroll-your-site-in-an-origin-trial).
+
+
+<!-- ====================================================================== -->
+## Define widgets
+
+widgets are defined in your PWA manifest file, by using the `widgets` manifest member. This manifest member is an array that can contain multiple widget definitions.
+
+```json
+{
+  "name": "PWAmp",
+  "description": "A music player app",
+  "icons": [
+    { "src": "img/icon-96.png", "sizes": "96x96" },
+    { "src": "img/icon-128.png", "sizes": "128x128" },
+    { "src": "img/icon-256.png", "sizes": "256x256" },
+    { "src": "img/icon-512.png", "sizes": "512x512" }
+  ],
+  "widgets": [
+    /* widget definitions go here */
+  ]
+}
+```
+
+Each entry in the `widgets` array contains several fields, as shown below:
+
+```json
+{
+  ...
+  "widgets": [
+    {
+      "name": "PWAmp mini player",
+      "description": "widget to control the PWAmp music player",
+      "tag": "pwamp",
+      "template": "pwamp-template",
+      "ms_ac_template": "widgets/mini-player-template.json",
+      "data": "widgets/mini-player-data.json",
+      "type": "application/json",
+      "screenshots": [
+        {
+          "src": "./screenshot-widget.png",
+          "sizes": "600x400",
+          "label": "The PWAmp mini-player widget"
+        }
+      ],
+      "icons": [
+        {
+          "src": "./favicon-16.png",
+          "sizes": "16x16"
+        }
+      ],
+      "auth": false,
+      "update": 100
+    }
+  ]
+}
+```
+
+In the above example, a music player application defines a mini player widget. The required and optional fields of a widget definition in the web app manifest are documented below:
+
+| Field | Description | Required |
+|:--- |:--- |:--- |
+| `name` | The title of the widget, presented to users. | Yes |
+| `short_name` | An alternative short version of the name. | No |
+| `description` | A description of what the widget does. | No |
+| `icons` | An array of icons to be used for the widget. | Yes |
+| `screenshots` | An array of screenshots that show what the widget looks like. Analogous to the [`screenshot` manifest member](https://developer.mozilla.org/docs/Web/Manifest/screenshots). | Yes |
+| `tag` | A string used to reference the widget in the PWA service worker. | Yes |
+| `template` | The template to use to display the widget in the operating system widgets dashboard. Note: although `template` is a required field, its value is currently not used. See `ms_ac_template` below. | Yes |
+| `ms_ac_template` | The URL to the custom Adaptive Cards template to use to display the widget in the operating system widgets dashboard. See [Define a widget template](#define-a-widget-template) below. | Yes |
+| `data` | The URL where the data to fill the template with can be found. For now, this URL is required to return valid JSON. | Yes |
```

Comment:
```suggestion
| `data` | The URL where the data to fill the template with can be found. This URL is required to return valid JSON. | Yes |
```
Justification: "For now, " is understood globally. Every sentence could say "For now, ", so is unneeded and can leverage terse brevity as might be appropriate for this particular technical doc.
https://developers.google.com/style/timeless-documentation
Avoid needless promising of future functionality throughout regular docs, if that principle applies to the present article.